lib/https.js

```diff
@@ -53,25 +53,31 @@ let debug = require('internal/util/debuglog').debuglog('https', (fn) => {
   debug = fn;
 });
 const { URL, urlToHttpOptions, searchParamsSymbol } = require('internal/url');
+const { validateObject } = require('internal/validators');
 
 function Server(opts, requestListener) {
   if (!(this instanceof Server)) return new Server(opts, requestListener);
 
   if (typeof opts === 'function') {
     requestListener = opts;
-    opts = undefined;
-  }
-  opts = { ...opts };
-
-  if (!opts.ALPNProtocols) {
-    // http/1.0 is not defined as Protocol IDs in IANA
-    // https://www.iana.org/assignments/tls-extensiontype-values
-    //       /tls-extensiontype-values.xhtml#alpn-protocol-ids
-    opts.ALPNProtocols = ['http/1.1'];
+    opts = kEmptyObject;
+  } else if (opts == null) {
+    opts = kEmptyObject;
+  } else {
+    validateObject(opts, 'options');
   }
 
   FunctionPrototypeCall(storeHTTPOptions, this, opts);
-  FunctionPrototypeCall(tls.Server, this, opts, _connectionListener);
+  FunctionPrototypeCall(tls.Server, this,
+                        {
+                          noDelay: true,
+                          // http/1.0 is not defined as Protocol IDs in IANA
+                          // https://www.iana.org/assignments/tls-extensiontype-values
+                          //       /tls-extensiontype-values.xhtml#alpn-protocol-ids
+                          ALPNProtocols: ['http/1.1'],
+                          ...opts,
```

Comment:
Nit: to keep the original behavior these should be swapped, otherwise the value of the `ALPNProtocols` can be overridden.

Reply:
As I understand, this is intended because `['http/1.1']` have to be used as default value of `ALPNProtocols` if `opts` don't have `ALPNProtocols`. Please let me know if I am wrong.

Reply:
You are correct. I misread the original code.